src/main/java/org/sagebionetworks/bridge/exporter3/Exporter3WorkerProcessor.java

```diff
@@ -528,6 +554,89 @@ private String getRawS3KeyForUpload(String appId, String studyId, Upload upload,
         return builder.toString();
     }
 
+    UploadTableRow getUploadTableRow(Upload upload, HealthDataRecordEx3 record, String recordId, Map<String, String> metadataMap )
+            throws IOException {
+        UploadTableRow tableRow = null;
+        String instanceGuid = metadataMap.get(METADATA_KEY_INSTANCE_GUID);
+        String assessmentGuid = metadataMap.get("assessmentGuid");
+        String appID = record.getAppId();
+
+        if (instanceGuid != null) {
+            tableRow = new UploadTableRow();
+            tableRow.setAssessmentGuid(metadataMap.get("assessmentGuid"));
+            String startedOn = metadataMap.get("startedOn");
+            DateTime createdOn = upload.getRequestedOn();
+            if (!Strings.isNullOrEmpty(startedOn)) {
+                createdOn = DateTime.parse(metadataMap.get("startedOn"));
+            }
+            tableRow.setCreatedOn(createdOn);
+            tableRow.setRecordId(recordId);
+            tableRow.setHealthCode(record.getHealthCode());
+            tableRow.setParticipantVersion(record.getParticipantVersion());
+            Map<String, String> metaData = new HashMap<>();
+            metaData.put("clientInfo", record.getClientInfo());
+            String[] metadataKeys = new String[] {
+                    "sessionGuid", "sessionStartEventId", "sessionStartEventTimestamp", "sessionStartEventTimestamp",
+                    "startedOn", "finishedOn", "declined", "sessionName"
+            };
+            for (String key: metadataKeys) {
+                String value = metadataMap.get(key);
+                if (value != null) {
+                    metaData.put(key, value);
+                }
+            }
+            tableRow.setMetadata(metaData);
+
+            boolean isTestData = false;
+            HealthDataRecord healthData = upload.getHealthData();
+            if (healthData != null) {
+                isTestData = healthData.getUserDataGroups().contains("test_user");
+                metaData.put("externalId", healthData.getUserExternalId());
+                metaData.put("dataGroups", String.join(",", healthData.getUserDataGroups()));
+            }
+            tableRow.setTestData(isTestData);
+
+            Assessment assessment = bridgeHelper.getAssessment(appID, assessmentGuid);
+            AssessmentConfig assessmentConfig = bridgeHelper.getAssessmentConfig(appID, assessmentGuid);
+            AssessmentSummarizerProvider summarizerProvider = new AssessmentSummarizerProvider();
+            AssessmentSummarizer summarizer = summarizerProvider.getSummarizer(assessment, assessmentConfig);
+            if (summarizer != null) {
+                File tempDir = null;
+                try {
+                    tempDir = fileHelper.createTempDir();
+                    File downloadedFile = fileHelper.newFile(tempDir, upload.getUploadId());
+                    s3Helper.downloadS3File(rawHealthDataBucket, upload.getUploadId(), downloadedFile);
```

Comment:
The filename won't just be upload ID. You'll need to call getRawS3KeyForUpload() to get the S3 Key.

There's also a weird edge case where EX3 isn't enabled for the app but is inabled for the study, in which case you won't have the information you need to download the file. This won't affect Open Bridge, but could affect other apps in the future. We should file a Jira to track this even if we choose not to handle it for 2023.